openmdao/core/driver.py

```diff
@@ -1687,6 +1688,289 @@ def _update_result(self, result):
+            assume_dvs_active is True, and the design variables are
+            returned as if they are on an active equality constraint.
+            These are the active indices _of_ the design var indices.
+        """
+        active_cons = {}
+        active_dvs = {}
+        des_vars = self._designvars
+        constraints = self._cons
+
+        # We obtain the driver scaled values so that feasibility check is performed
+        # with driver scaling.
+        dv_vals = self.get_design_var_values(driver_scaling=True)
+        con_vals = self.get_constraint_values(driver_scaling=True)
+
+        for constraint, con_options in constraints.items():
+            constraint_value = con_vals[constraint]
+            con_size = con_options['size']
+
+            if con_options.get('equals', None) is not None:
+                # Equality constraint, all indices active
+                active_cons[constraint] = {'indices': np.arange(con_size, dtype=int),
+                                           'active_bounds': np.zeros(con_size, dtype=int)}
+            else:
+                # Inequality constraint, determine active indices and bounds
+                constraint_upper = con_options.get("upper", np.inf)
+                constraint_lower = con_options.get("lower", -np.inf)
+
+                if np.all(np.isinf(constraint_upper)):
+                    upper_idxs = np.empty()
+                else:
+                    upper_mask = np.logical_or(constraint_value > constraint_upper,
+                                               np.isclose(constraint_value, constraint_upper,
+                                                          atol=feas_atol, rtol=feas_rtol))
+                    upper_idxs = np.where(upper_mask)[0]
+
+                if np.all(np.isinf(constraint_lower)):
+                    lower_idxs = np.empty()
+                else:
+                    lower_mask = np.logical_or(constraint_value < constraint_lower,
+                                               np.isclose(constraint_value, constraint_lower,
+                                                          atol=feas_atol, rtol=feas_rtol))
+                    lower_idxs = np.where(lower_mask)[0]
+
+                active_idxs = sorted(np.concatenate((upper_idxs, lower_idxs)))
+                active_bounds = [1 if idx in upper_idxs else -1 for idx in active_idxs]
+                if active_idxs:
+                    active_cons[constraint] = {'indices': active_idxs,
+                                               'active_bounds': active_bounds}
+
+        for des_var, des_var_options in des_vars.items():
+            des_var_value = dv_vals[des_var]
+            des_var_size = des_var_options['size']
+            des_var_upper = np.ravel(des_var_options.get("upper", np.inf))
+            des_var_lower = np.ravel(des_var_options.get("lower", -np.inf))
+
+            if assume_dvs_active:
+                active_dvs[des_var] = {'indices': np.arange(des_var_size, dtype=int),
```

Comment:
There is a weird corner case where both upper and lower can be active. It looks like this is supported by this overall method. There is one weird byproduct when assume_dvs_active is True, where the active bounds are all "0", which means equality constraint instead of containing both lower and upper. I am not sure if this matters or has any repercussions.